recipes/daggy/2.1/conanfile.py

```diff
@@ -1,138 +1,135 @@
-                    self.settings.compiler.version))
-
-        if not self.options["qt"].shared: 
-            raise ConanInvalidConfiguration("Shared Qt lib is required.") 
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+        self.options["qt"].shared = True
+        if conan_version.major == 1 and is_msvc(self) and self.options.shared:
+            # test_package fails with libssh2 linker errors:
+            # Ssh2Client.obj : error LNK2019: unresolved external symbol __imp_libssh2_exit
+            self.options.with_ssh2 = False
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
 
     def requirements(self):
-        self.requires("qt/6.2.2")
+        # Qt is used in the public headers
+        self.requires("qt/6.6.2", transitive_headers=True, transitive_libs=True)
         self.requires("kainjow-mustache/4.1")
-
         if self.options.with_yaml:
-            self.requires("yaml-cpp/0.7.0")
-
+            self.requires("yaml-cpp/0.8.0")
         if self.options.with_ssh2:
-            self.requires("libssh2/1.10.0")
+            self.requires("libssh2/1.11.0")
 
-    def _configure(self):
-        if self._cmake:
-            return self._cmake
+    def validate(self):
+        if self.settings.compiler.cppstd:
+            check_min_cppstd(self, self._min_cppstd)
+        minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
+        if minimum_version and Version(self.settings.compiler.version) < minimum_version:
+            raise ConanInvalidConfiguration(
+                f"{self.ref} requires C++{self._min_cppstd}, which your compiler does not support."
+            )
+        if not self.dependencies["qt"].options.shared:
+            raise ConanInvalidConfiguration("Shared Qt lib is required.")
 
-        self._cmake = CMake(self)
-        self._cmake.definitions["SSH2_SUPPORT"] = self.options.with_ssh2
-        self._cmake.definitions["YAML_SUPPORT"] = self.options.with_yaml
-        self._cmake.definitions["CONSOLE"] = self.options.with_console
-        self._cmake.definitions["PACKAGE_DEPS"] = False
-        self._cmake.definitions["VERSION"] = self.version
-        self._cmake.definitions["CONAN_BUILD"] = True
-        self._cmake.definitions["BUILD_TESTING"] = False
+    def build_requirements(self):
+        self.tool_requires("cmake/[>=3.21 <4]")
+        self.tool_requires("qt/<host_version>")
 
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def generate(self):
+        env = VirtualBuildEnv(self)
+        env.generate()
+
+        # Required for Qt's moc
+        env = VirtualRunEnv(self)
+        env.generate(scope="build")
+
+        tc = CMakeToolchain(self)
+        tc.variables["CMAKE_CXX_STANDARD"] = 17
```

Comment:
This seems a bit weird. Why is it forcing C++ 17? If this were necessary for the package, we should either add max_checks in the validate or erase the standard in the package_info methods